src/main/scala/chisel3/ModuleChoice.scala

```diff
@@ -0,0 +1,95 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chisel3
+
+import scala.language.experimental.macros
+import scala.collection.immutable.ListMap
+
+import chisel3.FixedIOBaseModule
+import chisel3.experimental.{BaseModule, SourceInfo}
+import chisel3.internal.{groupByIntoSeq, Builder, WireBinding}
+import chisel3.internal.Builder.pushCommand
+import chisel3.internal.firrtl.DefInstanceChoice
+import chisel3.internal.sourceinfo.InstChoiceTransform
+
+object ModuleChoice extends SourceInfoDoc {
```

Comment:
I think one other thing was to put this inside of `object choice` and just call it `Module`, then then general use will be `import chisel3.choice` and `choice.Module(...)`. Is this what y'all were thinking @azidar and @seldridge?

Reply:
I don't want to overload the `Module` name, I think we should have a different one, even if it's in its own package.